docs/apache-airflow/installation/installing-from-pypi.rst

```diff
@@ -79,50 +111,124 @@ where:
+packages to install and pin it to the version that you have, otherwise you might end up with a
+different version of Airflow than you expect as ``pip`` can upgrade/downgrade it automatically when
+performing dependency resolution.
+
+
+.. code-block:: bash
+
+    pip install "apache-airflow[celery]==|version|" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-|version|/constraints-3.8.txt"
+    pip install "apache-airflow==|version|" apache-airflow-providers-google==10.1.1
 
-Fixing Constraint files at release time
-'''''''''''''''''''''''''''''''''''''''
+You can also downgrade or upgrade other dependencies this way - even if they are not compatible with
+those dependencies that are stored in the original constraints file:
+
+.. code-block:: bash
+
+    pip install "apache-airflow[celery]==|version|" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-|version|/constraints-3.8.txt"
+    pip install "apache-airflow[celery]==|version|" dbt-core==0.20.0
+
+.. warning::
+
+    Not all dependencies can be installed this way - you might have dependencies conflicting with basic
+    requirements of Airflow or other dependencies installed in your system. However, by skipping constraints
+    when you install or upgrade dependencies, you give ``pip`` a chance to resolve the conflicts for you,
+    while keeping dependencies within the limits that Apache Airflow, providers and other dependencies require.
+    The resulting combination of those dependencies might not be tested together before as well as the set
+    of dependencies that come with constraints, but it should work in most cases as we usually add
+    requirements, when Airflow depends on particular versions of some dependencies. In cases you cannot
+    install some dependencies in the same environment as Airflow - you can attempt to use other approaches.
+    See :ref:`best practices for handling conflicting/complex Python dependencies <best_practices/handling_conflicting_complex_python_dependencies>`
+
+
+Verifying installed dependencies
+================================
+
+You can also always run the ``pip check`` command to test if the set of your Python packages is
+consistent and not conflicting.
+
+
+.. code-block:: bash
+
+    > pip check
+    No broken requirements found.
+
+
+When you see such message and the exit code from ``pip check`` is 0, you can be sure, that there are no
+conflicting dependencies in your environment.
+
+
+Using your own constraints
+==========================
+
+When you decide to install your own dependencies, or want to upgrade or downgrade providers, you might want
+to continue being able to keep reproducible installation of Airflow and those dependencies via a single command. In order to do that, you can produce your own constraints file and use it
+to install Airflow instead of the one provided by the community.
```

Comment:
```suggestion
to continue being able to keep reproducible installation of Airflow and those dependencies via a single command.
In order to do that, you can produce your own constraints file and use it to install Airflow instead of the 
one provided by the community.
```

Just suggesting a new line char as I see this line is relatively longer than all other lines :)